src/monitors/rocketMonitor.cpp

```diff
@@ -126,30 +146,71 @@ PetscErrorCode ablate::monitors::RocketMonitor::OutputRocket(TS ts, PetscInt ste
                             };
                             thrustTotal[d] = thrustTotal[d] + thrustCell[d];  // summation of total trust along fieldBoundary
                         }
+                        // update min and max cell pressure
+                        if (cellPressure < pressureMin[0]) {
+                            pressureMin[0] = cellPressure;
+                        }
+                        if (cellPressure > pressureMax[0]) {
+                            pressureMax[0] = cellPressure;
+                        }
+                        // update cell pressure sum for mean calculation
+                        pressureTotal[0] += cellPressure;
+
+                        // calculate cell mach number
+                        cellVelocity = PetscSqrtReal(PetscSqr(cellEuler[finiteVolume::CompressibleFlowFields::RHOU]) + PetscSqr(cellEuler[finiteVolume::CompressibleFlowFields::RHOU + 1]) +
+                                                     PetscSqr(cellEuler[finiteVolume::CompressibleFlowFields::RHOU + 2]));
+                        monitor->computeSpeedOfSound.function(conservedValues, &cellSpeedOfSound, monitor->computeSpeedOfSound.context.get());
+                        machNumber = cellVelocity / cellSpeedOfSound;
+
+                        // update min and max mach number
+                        if (machNumber < machMin[0]) {
+                            machMin[0] = machNumber;
+                        }
+                        if (machNumber > machMax[0]) {
+                            machMax[0] = machNumber;
+                        }
+                        // update mach number sum for mean calculation
+                        machTotal[0] += machNumber;
+                        numCells[0]++;  // Track number of cells on face in region of interest
                     }
                 }
             }
         }
 
         // Take across all ranks
+        MPI_Reduce(pressureMin, pressureMinGlob, 1, MPI_DOUBLE, MPI_MIN, 0, comm);
```

Comment:
These are a lot of calls to MPI_Reduce.  We might want to instead create a few arrays and store pressure, mach ... into a single array.